### PR TITLE
Fix #2891: posixlib spawn is now implemented.

### LINKED ISSUE #2891: Posixlib is missing spawn.h
POSIX spawn.h contains the definitions for `posix_spawn()`.

  `vfork()` has become deprecated in SN 0.5.0. macOS stopped supporting it in Monterrey 12.0 (?).
Thhe published macOS recommendation is to consider `posix_spawn()`. This is also mentioned in
the deprecation message.  Given this, more people, will be looking to use `posix_spawn()`

It would be pleasant & welcoming for these people to find it implemented.

More particularly, having `posix_spawnp()` implemented is a necessary, but not sufficient,
condition for javalib `Process` to gain back some of the speed it lost due to the recent
change away from `vfork()` to the much slower `fork()`.   Someday someone is going
to notice that slow down and be very unhappy. I hope Arman or David are not reading this, one
of my secrets is out.

### PR DESCRIPTION
**Pull request by LeeTibbert**
Fix #2891: posixlib spawn.scala & supporting spawn.c are now implemented.

Normally, one would expect a corresponding `spawnTest.scala`.

TL;DR; It is not good for tests to hang forever and `pselect()` and `waitid()` are not 
currently implemented in Scala Native.

### CODE REVIEW

**Comment by LeeTibbert** [comment]
I checked the log file and the failure has nothing direct to do with the change of this PR.
Because the test for posix_spawn requires a couple of other PRs, not code from 
this PR is ever executed. 

The error was on macOS, so it is likely that something, a child exit or a socket I/O is
getting dropped somewhere in the testing framework. I suspect the test framework
will need a 'supervisor' thread to go off once an hour or so to enumerate the tests
for which it is waiting.

**Review by WojciechMazur — APPROVED** [review]

File: posixlib/src/main/scala/scala/scalanative/posix/spawn.scala [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+  /* posix_spawnattr_t & posix_spawn_file_actions_t are opaque bulk storage
+   * types. They have no user accessible fields, not even the component Bytes.
+   * Users of these types should leave them opaque: i.e. no read, no write.
+   *
+   * The sizes here are from Linux 6.0.  Code in spawn.c checks at compile-time
+   * that these sizes are greater than or equal to the equivalent OS types.
+   * Maintainers:  If you change sizes, either here or in spawn.c, change
+   *               the other file also.
+   */
```
[the remark this answers is not in the document]
Comment (LeeTibbert) [reply posixlib/src/main/scala/scala/scalanative/posix/spawn.scala:27]: @ekrich Thank you for the review & the suggestion.

I have let your suggestion brew on my back brain for several hours.  I come
back to my original thoughts:  These are developer focused comments and
should not be Scaladoc. 

I am a great believer in doc starting with a best guess "appropriate precision" and
supplying easy ways to get both less (dictionaries & tutorials) or more (read the source)
information.  Finding that sweet spot is art and sometimes skill.

We both have struggled, or at least I have, with reading and interpreting the Open Group POSIX
specification.  Sometimes implementation depends on a particular reading of the specification.
The first paragraph is stating my understanding of what the specification means when it does
not give structure members or layout.  In other places these are referred to as 'opaque'.
I was trying to give _developers_ a hint beyond a mutable `Array[Byte]` why they were not
seeing a corresponding `struct` declaration.   This implementation/interpretation detail should
not be hidden from an end user: they can read the source.  Since it does not affect their usage
of types described by POSIX, it is has negative value (i.e. source of confusion) at the Scaladoc level.

Certainly the second paragraph is "developer only". It is intended to teach cross ties to 
"new" or returning-after-a-long-hiatus maintainers. 

Based on your being an independent set of eyes looking at the comment, can you suggest
alternate Scaladoc wording for the first comment?

**Review by WojciechMazur — APPROVED** [review]
